.github/workflows/test-pr.yml

```diff
@@ -1,17 +1,17 @@
 name: CodeClimate-Checks
 on:
   push:
-    branches-ignore: 
+    branches:
       - 'develop'
-      - 'test'
-      - 'master'
+      - '!test'
```

Comment:
this syntax seems off

Reply:
Since what is the point of listing develop here if you have to implicitly list the !branches. If this by default includes all branches just stick with doing ignore.

Reply:
We need to use `!` to exclude branches if we are using `branches` (which we are, to target develop) we can't use `branches-ignore` under the same event.

Reply:
If we only do branches ignore and add `master` and `test`, then the runs will be triggered twice for each push to a PR, one for `push` and one for `pull_request`

Reply:
I fail to see how functionally these two are different.
```
branches:
   - 'develop'
   - '!test'
   - '!master'
```
```
branches-ignore:
   - 'test'
   - 'master'
```

Reply:
Since if you are saying you need to target  develop wouldn't just this suffice
```
branches:
   - 'develop'
```

Reply:
To reply to the first comment:

```
branches:
   - 'develop'
   - '!test'
   - '!master'
```
   
```
branches-ignore:
   - 'test'
   - 'master'
```

the second block will trigger the event twice every time the PR is updated.

```
branches:
   - 'develop'
```
   
With this one, when we merge develop to test and master, the checks will run.